server/resp/src/main/java/org/infinispan/server/resp/Intrinsics.java

```diff
@@ -0,0 +1,145 @@
+package org.infinispan.server.resp;
+
+import java.nio.charset.StandardCharsets;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.ByteProcessor;
+
+public class Intrinsics {
+   private static final int TERMINATOR_LENGTH = 2;
+
+   public static byte singleByte(ByteBuf buffer) {
+      if (buffer.isReadable()) {
+         return buffer.readByte();
+      } else return 0;
+   }
+
+   public static String simpleString(ByteBuf buf) {
+      // Find the end LF (would be nice to do this without iterating twice)
+      int offset = buf.forEachByte(ByteProcessor.FIND_LF);
+      if (offset <= 0 || buf.getByte(offset - 1) != '\r') {
+         return null;
+      }
+      String simpleString = buf.toString(buf.readerIndex(), offset - 1 - buf.readerIndex(), StandardCharsets.US_ASCII);
+      // toString doesn't move the reader index forward
+      buf.readerIndex(offset + 1);
+      return simpleString;
+   }
+
+   public static long readNumber(ByteBuf buf, Resp2LongProcessor longProcessor) {
+      long value = longProcessor.getValue(buf);
+      // The longProcessor can technically read up to only /r, we need to ensure there is /r and /n
+      if (longProcessor.complete && buf.readableBytes() >= longProcessor.bytesRead + TERMINATOR_LENGTH) {
+         buf.skipBytes(longProcessor.bytesRead + TERMINATOR_LENGTH);
+      }
+      return value;
+   }
+
+   public static byte[] readTerminatedBytes(ByteBuf buf) {
+      // Find the end LF (would be nice to do this without iterating twice)
+      int offset = buf.forEachByte(ByteProcessor.FIND_LF);
+      if (offset <= 0 || buf.getByte(offset - 1) != '\r') {
+         return null;
+      }
+      byte[] data = new byte[offset - 1 - buf.readerIndex()];
+      buf.readBytes(data);
+      buf.skipBytes(TERMINATOR_LENGTH);
+      return data;
+   }
+
+   public static String bulkString(ByteBuf buf, Resp2LongProcessor longProcessor) {
+      buf.markReaderIndex();
+      long longSize = readNumber(buf, longProcessor);
+      if (longSize > Integer.MAX_VALUE) {
+         throw new IllegalArgumentException("Bytes cannot be longer than " + Integer.MAX_VALUE);
+      }
+      if (longSize == Long.MIN_VALUE) {
+         return null;
+      }
```

Comment:
I didn't get why this one. Should it be `!longProcessor.complete`, or is it just because of the int cast?

Reply:
Ah, yes this was a left over before I changed the processor, it should check the reader index instead to determine if something was read.

Reply:
The `bulkArray` also needs this.